### PR TITLE
UCP: Group protocol thresholds in ucp_worker, and print them in ucx_info.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1437/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/824/ for details (Mellanox internal link).

**Comment by MattBBaker** [comment]
autobot build

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -538,6 +538,22 @@ void ucp_worker_destroy(ucp_worker_h worker);
 
 
 /**
+ * @ingroup UCP_CONTEXT
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: I think this belongs to the worker

File: src/ucp/api/ucp.h [hunk 2]
```diff
@@ -538,6 +538,22 @@ void ucp_worker_destroy(ucp_worker_h worker);
...
+ * @bried Print information about protocols that would be used by a worker.
+ *
+ * This routine prints information about the protocols being used, thresholds,
+ * UCT methods, and so on, used by the current context.
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: "UCT methods" -> transport methods
" and so on, used by the current context" -> " and other useful information associated with the worker"

File: src/ucp/api/ucp.h [hunk 3]
```diff
@@ -538,6 +538,22 @@ void ucp_worker_destroy(ucp_worker_h worker);
...
+ *
+ * @param [in] context      Handle to @ref ucp_context_h "UCP application context".
+ * @param [in] stream       Output stream to print the information to.
+ * @param [in] title        Configuration title to print.
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: hmm. do we need print or query interface ? Maybe we just need a query and user will print this out in whatever format it wants ?

Comment (yosefe) [inline src/ucp/api/ucp.h]: it was simpler to define just print interface than query interface for now, for ucx_info.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/828/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1441/ for details.

**Comment by shamisp** [comment]
:+1:

**Pull request merged**